### ISSUE TITLE
Support for SITEURL links to be recognised as "internal"

### ISSUE THREAD

**Issue by galaxy4public** [rlaboiss/pelican-linkclass#32]
Currently, the logic is purely based on whether the link has "https?://" as the string prefix for `href`.  However, there are quite a few sites that use `SITEURL` Pelican setting to define the full URL for the generated internal links (this is mostly needed to support things like the sitemap plugin and others).  When `SITEURL` is defined, this plugin would mark all links as external, which kind of renders it useless in these configurations.

**Comment by rlaboiss** [comment]
Thanks for your issue report, but I am afraid I do not understand what you mean. I maintain a Pelican-powered website that has `SITEURL` defined in `publishconf.py` and the linkclass plugin works perfectly for it.

The linkclass plugin is implemented as a Markdown extension, that just recognize these two distinct ways of defining hyperlinks:
```markdown
[This is an internal link]({filename}/another-file-in-this-site.md) 
and [this is an external link](https://outside-host.com).
```
and adds the appropriate `class` attribute to the generated HTML element.

You are perhaps making a wrong assumption about how the plugin works. It indeed looks for the pattern `https?://` in `href`, but it does it _before_ the Pelican engine generates the final URL, by inserting the prefix defined in `SITEURL`. 

Unless you come up with a concrete example of a Pelican website for which the linkclass plugin fails when `SITEURL` is defined, I will close this issue report.

**Comment by galaxy4public** [comment]
@rlaboiss , thanks for the extended response.  You are correct (and I need to do better in terms of opening issues after just reading through the code).  One thing I noticed though (after installing the plugin) is that the documented configuration does not match (and does not work) with the plugin.  It requires three arguments instead of two in each tupple.  Should I open an issue for this?

**Comment by rlaboiss** [comment]
> One thing I noticed though (after installing the plugin) is that the documented configuration does not match (and does not work) with the plugin. It requires three arguments instead of two in each tupple. Should I open an issue for this?

Yes, please, open an issue report for it, since this is indeed a bug in the documentation.

**Comment by rlaboiss** [comment]
> Yes, please, open an issue report for it, since this is indeed a bug in the documentation.

I just did it: #34

**Comment by rlaboiss** [comment]
The issue you raised is fixed in version 2.1.0 of the linkclass plugin (commit 36720e0).

Notice that the code has been adjusted, such that it corresponds to the previous documentation for the LINKCLASS variable (only two elements in each tuple).

Thanks for reporting the problem!
